loopy/target/c/codegen/expression.py

```diff
@@ -684,9 +684,11 @@ def map_min(self, expr, enclosing_prec):
     map_max = map_min
 
     def map_if(self, expr, enclosing_prec):
-        from pymbolic.mapper.stringifier import PREC_NONE
+        from pymbolic.mapper.stringifier import PREC_NONE, PREC_CALL
         return "({} ? {} : {})".format(
-                self.rec(expr.condition, PREC_NONE),
+                # Force parentheses around the condition to prevent compiler
+                # warnings regarding precedence.
```

Comment:
Please be specific, to allow verification that this is still needed.

Reply:
Ok, added this in f88c890c